primitives/state-machine/src/ext.rs

```diff
@@ -754,14 +736,14 @@ mod tests {
 		let mut overlay = prepare_overlay_with_changes();
 		let mut offchain_overlay = prepare_offchain_overlay_with_changes();
 		let mut cache = StorageTransactionCache::default();
-		overlay.prospective.top.get_mut(&vec![1]).unwrap().value = None;
+		overlay.set_storage(vec![1], None);
```

Comment:
This break the test, the test was testing that storage_changes_root_is_some_when_extrinsic_changes_are_empty , fwiu not registering the extrinsic change was intentional here.
Maybe this can be achieved by changes.set_collect_extrinsics(false); then set_storage then changes.set_collect_extrinsics(true);.
The root should not change.

Reply:
Good point. Done.